skimage/measure/__init__.py

```diff
@@ -7,7 +7,7 @@
 from ._pnpoly import points_in_poly, grid_points_in_poly
 from ._moments import moments, moments_central, moments_normalized, moments_hu
 from .profile import profile_line
-from .fit import LineModel, CircleModel, EllipseModel, ransac
+from .fit import LineModel, LineModel3D, CircleModel, EllipseModel, ransac
```

Comment:
Hmm. It would be simple enough to harmonize the outputs to (point, unit vector) - potentially adding an attribute (`old_params`?) or kwarg to regain the old behavior. In pursuit of a dramatically improved, generalizable, API I would support a breaking change.

Regarding other methods, it's easy enough to house all logic for this in a `._check(self, data)` method called from `estiamate` or `residuals` and populate the rest of the class, including optional methods. Particularly because it seems `predict_x` and `predict_y` are intended to be called after `estimate`. Alternatively, `LineModel` could simply not expose them but `LineModel2D` could retain them.